### PR TITLE
feat: buy with sell and repay

### PR DESCRIPTION
**Pull request by uzielflorida**

### CODE REVIEW

File: src/contracts/PurchaseBundler.ts [hunk 1]
```diff
@@ -1,24 +1,67 @@
...
+        type: 'tuple',
+        components: [
+          { name: 'module', type: 'address' },
+          { name: 'data', type: 'bytes' },
+          { name: 'value', type: 'uint256' },
+        ],
+      },
+      { name: 'contractMustBeOwner', type: 'bool' },
+    ],
+  } as const;
```
Comment (uzielflorida) [inline src/contracts/PurchaseBundler.ts:45]: This is sadly not in the ABI since it's not a param of anything

File: examples/buyWithSellAndRepay.ts
Comment (mishuagopian) [inline examples/buyWithSellAndRepay.ts:1]: bump beta version

File: src/deploys.ts [hunk 2]
```diff
@@ -18,7 +18,12 @@ interface Contracts {
     v5: Address;
     v6: Address;
   };
-  PurchaseBundler: Address;
+  PurchaseBundler: {
+    v6: {
+      mslv5: Address;
+      mslv6: Address;
+    };
```
Comment (mishuagopian) [inline src/deploys.ts:25]: remove this level

File: src/contracts/index.ts [hunk 3]
```diff
@@ -88,6 +101,21 @@ export class Contracts {
...
+   * @returns The corresponding AuctionLoanLiquidator contract
+   */
+  PurchaseBundler(contractAddress: Address) {
+    if (areSameAddress(contractAddress, this.MultiSourceLoanV5.address)) {
+      return this.PurchaseBundlerMslV5;
+    }
+    if (areSameAddress(contractAddress, this.MultiSourceLoanV6.address)) {
+      return this.PurchaseBundlerMslV6;
+    }
+    throw new Error(`Invalid Contract Address ${contractAddress}`);
```
Comment (mishuagopian) [inline src/contracts/index.ts:116]:
```suggestion
  PurchaseBundler(mslContractAddress: Address) {
    if (areSameAddress(mslContractAddress, this.MultiSourceLoanV5.address)) {
      return this.PurchaseBundlerMslV5;
    }
    if (areSameAddress(mslContractAddress, this.MultiSourceLoanV6.address)) {
      return this.PurchaseBundlerMslV6;
    }
    throw new Error(`Invalid Contract Address ${mslContractAddress}`);
```

File: src/contracts/index.ts [hunk 4]
```diff
@@ -88,6 +101,21 @@ export class Contracts {
     throw new Error(`Invalid Contract Address ${contractAddress}`);
   }
 
+  /**
+   *
+   * @param contractAddress The contract address of the MSL contract
+   * @returns The corresponding AuctionLoanLiquidator contract
```
Comment (mishuagopian) [inline src/contracts/index.ts:107]:
```suggestion
   * @param mslContractAddress The contract address of the MSL contract
   * @returns The corresponding PurchaseBundler contract
```

File: src/deploys.ts [hunk 5]
```diff
@@ -18,7 +18,10 @@ interface Contracts {
     v5: Address;
     v6: Address;
   };
-  PurchaseBundler: Address;
+  PurchaseBundler: {
+    mslv5: Address;
+    mslv6: Address;
```
Comment (mishuagopian) [inline src/deploys.ts:23]: remove msl

File: src/contracts/index.ts [hunk 6]
```diff
@@ -54,7 +58,16 @@ export class Contracts {
     this.AuctionLoanLiquidatorV6 = new AllV6({ walletClient });
     this.UserVaultV5 = new UserVaultV5({ walletClient });
     this.UserVaultV6 = new UserVaultV6({ walletClient });
-    this.PurchaseBundler = new PurchaseBundler({ walletClient });
+    this.PurchaseBundlerMslV5 = new PurchaseBundler({
+      walletClient,
+      contractAddress: PurchaseBundlerContract.mslv5,
+      msl: this.MultiSourceLoanV5,
+    });
+    this.PurchaseBundlerMslV6 = new PurchaseBundler({
```
Comment (mishuagopian) [inline src/contracts/index.ts:66]: remove msl

**Review by mishuagopian — APPROVED** [review]

**Pull request merged**